pycrostates/preprocessing/spatial_filter.py

```diff
@@ -0,0 +1,165 @@
+from typing import Union
+
+import mne
+import numpy as np
+from mne import BaseEpochs
+from mne.channels.interpolation import _make_interpolation_matrix
+from mne.io import BaseRaw
+from mne.io.pick import _picks_by_type
+from mne.parallel import parallel_func
+from mne.utils.check import _check_preload
+
+from ..utils._checks import _check_n_jobs, _check_type, _check_value
+from ..utils._docs import fill_doc
+from ..utils._logs import logger, verbose
+
+
+@fill_doc
+@verbose
+def apply_spatial_filter(
+    inst: Union[BaseRaw, BaseEpochs],
+    ch_type: str = "eeg",
+    exclude_bads: bool = True,
+    n_jobs: int = 1,
+    verbose=None,
+):
+    r"""Apply spatial filter.
+
+    Adapted from \ :footcite:p:`michel2019eeg`.
```

Comment:
`:p:` renders only as a numerical right?
IIRC, the other option, which I think would be more adapted to this sentence includes the author/year, `:n:`?